contracts/exchange/contracts/Exchange.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity ^0.6;
+
+// SPDX-License-Identifier: UNLICENSED
+
+import "openzeppelin-solidity/contracts/access/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/utils/Pausable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+import "./ExchangeERC20.sol";
+
+contract Exchange is ExchangeERC20, Ownable, Pausable {
+    using SafeMath for uint;
+    using SafeERC20 for IERC20;
+
+    uint256 public r; // A / B = r [A >= B]
+    address public assetA;
+    address public assetB;
+
+    event LiquidityMinted(address indexed by, uint amount, uint _a, uint _b);
+    event LiquidityBurned(address indexed by, uint amount, uint _a, uint _b);
+    event Swap(address indexed who, address indexed _assetA, address _assetB, uint _a, uint _b);
```

Comment:
I think you can simplify the swap event by 
`event Swap(address indexed who, uint _aAmount, uint _bAmount)`
when `_aAmount` will always refer to assetA

Reply:
Hi @leonprou ! So I can simplify it by saying just the swapped quantities one way and the other, right?
But it would not give the direction (from A to B | from B to A).
What if we emit two events `event SwapAforB(address indexed who, uint _aAmount, uint _bAmount)` and `event SwapBforA(address indexed who, uint _aAmount, uint _bAmount)` ?

Reply:
Ok I see. Yeah you're right on that one.
I think we can go with that implementation, no need to splitting the events (more events to look after).
Another small question, why `_assetA` is  `indexed`?

Reply:
This way you can search for the assets being swapped. For example:

If a swap from A to B happens an event `Swap(msg.sender, A, B, amountA, amountB)` is created.
If a swap from B to A happens an event `Swap(msg.sender, B, A, amountB, amountA)` is created.

This way you can search for A -> B swaps and the other way around.

Reply:
Also, I cleaned the repo from unused libraries.

Reply:
Need to index just by the first token as the knowing the first we know the second, got you 👍